webtau-http/src/test/groovy/com/twosigma/webtau/http/datanode/StructuredDataNodeTest.groovy

```diff
@@ -123,4 +123,75 @@ class StructuredDataNodeTest {
         List<Map<String, Object>> list = node.get('key1').get()
         list.should == [[name: 'name1'], [name: 'name2']]
     }
+
+    @Test
+    void "should access underlying value when provided path"() {
+        def node = DataNodeBuilder.fromMap(new DataNodeId("body"), [
+            key1: [name: 'name1'],
+            key2: [name: 'name2'],
+        ])
+
+        String name = node.get('key1.name').get()
+        name.should == 'name1'
+    }
+
+    @Test
+    void "should access array element via path"() {
+        def node = DataNodeBuilder.fromMap(new DataNodeId("body"), [
+            key1: [[name: 'name1'], [name: 'name2']]
+        ])
+
+        Map<String, String> firstElement = node.get('key1[0]').get()
+        firstElement.should == [name: 'name1']
+
+        String firstName = node.get('key1[0].name').get()
```

Comment:
how about case like `root.child[idx].anotherChild`?

Reply:
You mean using a var for the index?

Reply:
no, just a usecase where index is in the middle, not just start or end

Reply:
Oh, ok.  It should work but I'll add a test.